backend/app/poller.py

```
@@ -883,6 +858,198 @@ def get_pref(attr, default_attr):
+    # 3. Batch-fetch item counts for all relevant slots
+    counts_by_slot = {info['slot_id']: {} for info in evaluation_targets.values()}
+    slot_ids = list(counts_by_slot.keys())
+    if slot_ids:
+        try:
+            counts = session.query(SlotItemCount).filter(
+                SlotItemCount.room_id == room_uuid,
+                SlotItemCount.slot_id.in_(slot_ids)
+            ).all()
+            for c in counts:
+                counts_by_slot[c.slot_id][c.item_id] = c.count
+        except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)